refs #47766: Bugfixes

ZeeCapp opened the pull request:
- Windows max file path length bugfix
- Add first part of GUID to email naming to avoid conflicting email names
- Add HID to folder naming to avoid conflicting names

Commit:
refs #47766: Bugfixes

ZeeCapp committed:
Merge branch 'eway-crm:master' into master

rstefko commented on ApiConnection.cs:
```diff
@@ -69,16 +70,23 @@ public void DownloadEmail(Guid emailGuid, string directory)
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
-                Logger.LogDebug($"File '{directory}' already exist");
+                filePath = FileHelper.TrimPathLength(filePath, 260);
```
Lepší by bylo trimovat ten emailName, klidně na něco menšího (třeba 64 znaků?), v těch velkých názvech se asi stejně nikdo nevyznaná, jde o to, aby ti tam pak zůstala i ta unikátní část.

rstefko replied:
Jinak pokud to spouštíš jako admin, tak by ses měl dostat i k Server_ID, což je rostoucí řada (INT).

ZeeCapp replied:
@rstefko fixed, přidal jsem tam trim na file name i folder name, ale tenhle check jsem tam radši ještě nechal, ten script by mohl někdo spustit z nějakého deeply nested adresáře a potom by ten trim na 64 znaků v názvu nestačil. Nechal jsem to tam teda jen jako poslední záchranu.

Taky jsem dodělal přidávání první části GUID do názvu dokumentů, stejně jako to je u emailů, aby nedocházelo k nějakým falešným duplicitám kvůli stejnému názvu.

Commit:
refs #47766: PR Changes

- Add first part of GUID to document names to avoid false duplicates
- Add first part of GUID to folder names to avoid false duplicates
- Add a check for files ending in a dot on windows
